test/common/typeconfig.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
+	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetTypeConfig(dynClient client.Client, name, namespace string) (typeconfig.Interface, error) {
```

Comment:
(No action required) controller-runtime's client -> generic.  client-go's unstructured client -> dynamic.  I've managed to screw up the naming, but I'm assuming we'll end up using the generic client everywhere and won't need to differentiate at some point in the future.

Reply:
Updated to `dynamicClient`!